virttest/utils_v2v.py

```diff
@@ -1841,3 +1859,178 @@ def compare_version(interval, version=None, cmd=None):
+                    logging.debug(
+                        "Got username prompt; sending '%s'", username)
+                    session.sendline(username)
+                elif match in [2, 3, 4, 5]:
+                    logging.debug(
+                        "Got password prompt, sending '%s'",
+                        asterisk_passwd(password))
+                    session.sendline(password)
+                elif match == 6:  # Wait for custom input
+                    logging.debug(
+                        "Got console '%s', send input list %s", match, choices)
+                    session.sendline(choices)
+                elif match == 7:  # LUKS password
+                    logging.debug(
+                        "Got password prompt, sending '%s'",
+                        asterisk_passwd(luks_password))
+                    session.sendline(luks_password)
+                elif match == 8:  # Done
+                    break
+                elif match == 9:  # Interaction Finish
+                    timeout = running_timeout
+            except aexpect.ExpectTimeoutError:
+                # when free_running is true and timeout happens, it means
+                # the command doesn't have any response, may be dead or
+                # performance is quite poor.
+                logging.debug("timeout happens")
+                if free_running:
+                    raise
+
+                # If timeout happens two times and the last line are same,
+                # it means v2v may be dead or performance is quite poor.
+                cont = session.get_output()
+                new_last_line = ''
+                nonempty_lines = [
+                    l for l in cont.splitlines()[-10:] if l.strip()]
+                if nonempty_lines:
+                    new_last_line = nonempty_lines[-1]
+                if last_line and last_line == new_last_line:
+                    logging.debug(
+                        'v2v command may be dead or have bad performance')
+                    raise
+                last_line = new_last_line
+
+                # Set a big timeout value when interaction finishes
+                for pattern in FREE_RUNNING_PROMPTS:
+                    if re.search(pattern, cont):
+                        logging.debug(
+                            "interaction finished and begin running freely")
+                        free_running = True
+                        timeout = running_timeout
+                        break
+
+    try:
+        subproc = aexpect.Expect(*args, **kwargs)
+        logging.debug('Running command: %s', subproc.command)
+        handle_prompts(subproc, timeout)
+    except aexpect.ExpectProcessTerminatedError:
+        # v2v cmd is dead or finished
+        pass
+    except Exception:
+        # aexpect.ExpectTimeoutError or other exceptions
+        # send 'ctrl+c' to v2v process to interrupt running quickly
+        subproc.sendcontrol('c')
+        raise
```

Comment:
I see some 'raise's here , do we need to raise something?

Reply:
Yes, we need to raise them again after capturing. and they are helpful for finding script errors.

Reply:
So we do not need to raise specific exceptions here, but we will raise them later, as they are all the same kind of errors?

Reply:
I don't think so.  Because the raw exception is enough. The 'raise' here is mainly for checking the backtrace of an unexpected exception.

Reply:
Then I'm confused, can I put this way: in this function, all these errors are considered as the same exception?

Reply:
Not quite understand your confusion. 
aexpect.ExpectProcessTerminatedError  is expected, it means v2v finished or v2v process is dead. We treat it as success.
Exception, this is mainly used to capture 'aexpect.ExpectTimeoutError', then kill v2v process in advance because Aexcpect server will wait the client process to finish.  In the meanwhile, you may have other excpetion when your changed something, like undefine variants, etc. Those exceptions should also be captured as 'aexpect.ExpectTimeoutError' because of same reason.
If the excpetion is not raised again, the bacetrace will not be printed. It's hard for debugging a typing error or something else.

Reply:
Then I understand, thank you for explaining.

Reply:
I guess it's ok to me.